packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -960,25 +789,13 @@ public ResponseEntity<JsonNode> removeProjectGroupPermissions(
 	}
 
 	@Operation(summary = "Toggle a project public, or restricted, by ID")
-	@ApiResponses(
-			value = {
-				@ApiResponse(
-						responseCode = "200",
-						description = "Project visibility has been updated",
-						content = {
-							@Content(
-									mediaType = "application/json",
-									schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = UUID.class))
-						}),
-				@ApiResponse(
-						responseCode = "304",
-						description = "The current user does not have privileges to modify this project.",
-						content = @Content),
-				@ApiResponse(
-						responseCode = "500",
-						description = "An error occurred verifying permissions",
-						content = @Content)
-			})
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Project visibility has been updated", content = {
+					@Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = UUID.class))
+			}),
+			@ApiResponse(responseCode = "304", description = "The current user does not have privileges to modify this project.", content = @Content),
+			@ApiResponse(responseCode = "500", description = "An error occurred verifying permissions", content = @Content)
+	})
 	@PutMapping("/set-public/{id}/{isPublic}")
 	@Secured(Roles.USER)
 	public ResponseEntity<JsonNode> makeProjectPublic(
```

Comment:
I would have thought this would propagate the isAssetPublic logic through to its children, either setting to true or false as appropriate

Reply:
Oh lord so the flag doesn't even get set on the underlying project. Okay I'll inject some code here too.